ee/tabby-schema/src/schema/web_documents.rs

```diff
@@ -0,0 +1,81 @@
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use juniper::{GraphQLInputObject, GraphQLObject, ID};
+use validator::Validate;
+
+use crate::{job::JobInfo, juniper::relay::NodeType, Context, Result};
+
+#[derive(GraphQLObject)]
+#[graphql(context = Context)]
+pub struct CustomWebDocument {
+    pub url: String,
+    pub name: String,
+    pub id: ID,
+    pub created_at: DateTime<Utc>,
+    pub job_info: JobInfo,
+}
+
+#[derive(GraphQLObject)]
+#[graphql(context = Context)]
+pub struct PresetWebDocument {
+    pub name: String,
+    pub id: ID,
+    pub job_info: JobInfo,
+}
+
+impl CustomWebDocument {
+    pub fn source_id(&self) -> String {
+        Self::format_source_id(&self.id)
+    }
+
+    pub fn format_source_id(id: &ID) -> String {
+        format!("web_document:{}", id)
+    }
+}
+
+#[derive(Validate, GraphQLInputObject)]
+pub struct CreateCustomDocumentInput {
+    #[validate(url(code = "name", message = "Invalid Name"))]
+    pub name: String,
```

Comment:
why use url validator for name?
https://github.com/Keats/validator?tab=readme-ov-file#url

For name validation, consider following examples like: https://demo.tabbyml.com/files/github/TabbyML/tabby/-/blob/b559217f6f48cdbedd1748919c61826a917d3517/ee/tabby-schema/src/schema/constants.rs?plain=1#L4

Reply:
Ok, I have refactor this defination.